src/Elastic.Apm/Api/Context.cs

```diff
@@ -4,25 +4,29 @@
 
 using System;
 using System.Collections.Generic;
+using Elastic.Apm.Model;
 using Elastic.Apm.Report.Serialization;
 using Newtonsoft.Json;
 
 namespace Elastic.Apm.Api
 {
 	public class Context
 	{
-		private readonly Lazy<Dictionary<string, string>> _custom = new Lazy<Dictionary<string, string>>();
-		private readonly Lazy<Dictionary<string, string>> _labels = new Lazy<Dictionary<string, string>>();
+		private Lazy<Dictionary<string, string>> _custom = new Lazy<Dictionary<string, string>>();
 
 		[JsonConverter(typeof(CustomJsonConverter))]
 		public Dictionary<string, string> Custom => _custom.Value;
 
+		internal Lazy<LabelsDictionary> InternalLabels = new Lazy<LabelsDictionary>();
+
 		/// <summary>
 		/// <seealso cref="ShouldSerializeLabels" />
 		/// </summary>
 		[JsonProperty("tags")]
 		[JsonConverter(typeof(LabelsJsonConverter))]
```

Comment:
I don't think this should have converter type `LabelsJsonConverter`. The `[JsonConverter(typeof(LabelsJsonConverter))]` can be applied to `LabelsDictionary` and removed from label properties on `Context`, `Metadata`, `SpanContext` e.g.

```c#
	[JsonConverter(typeof(LabelsJsonConverter))]
	internal class LabelsDictionary : Dictionary<string, string>
    {
        // ...
    }
```

It achieves the same outcome and removes the need to attribute every labels property

Reply:
Makes sense - I implemented it. Only thing is that if we do this then we'll serialize labels through the `InternalLabels` property instead of the `Labels` property and `InternalLabels` is `Lazy<LabelsDictionary>`  which again needs another converter. But overall I think it's still better that way.

Reply:
I think we might be talking about slightly different things. I suggested to move the `[JsonConverter(typeof(LabelsJsonConverter))]` on all label properties to the `LabelsDictionary` type. With this change, wherever `LabelsDictionary` is used, the converter will be used. I _think_ you're talking about attributing `InternalLabels`, which I don't think we should do, or need to do. The public API of the `"tags"` property is still `Dictionary<string, string>`, but because the specific implementation of the instance returned is `LabelsDictionary` that has an associated `JsonConverter`, the converter will be used. I think it's a little unituitive to be honest, but allows keeping the lazy semantics and moves the association of `LabelsDictionary` and `LabelsJsonConverter` to one place.

Reply:
Oh, I think now I got it. I just pushed a commit: https://github.com/elastic/apm-agent-dotnet/pull/982/commits/c1eb1680418b85f7b193cce051aebaa6e3082917 

Is that what you mean? 

I kind of agree it's a bit unituitive and the other thing is that we'll serialize labels through an obsolete property. But overall I'm fine with it. Plus I agree it's nice to have the association of `LabelsDictionary` and `LabelsJsonConverter` at one place